specificatie/klanten/openapi.yaml

```diff
@@ -1046,8 +1084,16 @@ components:
           type: string
           maxLength: 3
           example: "nld"
-        adres:
-          $ref: '#/components/schemas/KlantAdres'
+        correspondentieadres:
+          oneOf:
```

Comment:
De ervaringen bij Haal centraal hebben geleerd dat 'oneOf' bij codegeneratie voor problemen zorgt.
Bij ZGW is deze wijze toch ook niet gebruikt, daar is toch ook gebruik gemaakt van een discriminator of vergis ik me dan?

Voorstel om het daarom als volgt te doen:

```
    correspondentieadres:
      $ref: #/components/schemas/AbstractCorrespondentieadres
    ...

    AbstractCorrespondentieadres
      type: object
      description: |
        Gegevens over het verblijf of de woonlocatie van een persoon.
      required:
        - type
      properties:
        type:
          type: string
      discriminator:
        propertyName: type
        mapping:
          Binnenlandsadres: '#/components/schemas/Binnenlandsadres'
          Buitenlandsadres: '#/components/schemas/Buitenlandsadres'
          Postadres: '#/components/schemas/Postadres'

```

Reply:
Ja, zie subjectType. Consistentie lijkt me belangrijk; tenzij er hele goede argumenten vóór 'oneOf' zijn zou ik dit inderdada op de 'bekende' wijze oplossen.

Reply:
Aanvankelijk wilde ik het op de zelfde manier oplossen als bij `subjectType` (oftewel de `discriminator`-constructie). Echter dan krijg je te maken met geneste discriminators. Dat heb ik nog nooit eerder gezien en de nesting werd zelfs zo ingewikkeld dat ik er zo snel even niet uitkwam en heb maar gekozen voor de `oneOf`-constructie. Die werkt trouwens in tegenstelling tot `discriminator` zowel in Swagger als in Redoc. Mocht het uiteindelijk toch mogelijk zijn om zonder teveel poespas een geneste discriminator te specificeren dan heb ik daar zeker geen bezwaar tegen.

Reply:
Dank Robert voor de tip, ik heb dit overgenomen in de nieuwe versie.